src/main/java/net/geforcemods/securitycraft/entity/SentryEntity.java

```diff
@@ -186,14 +201,26 @@ else if(item == SCContent.WHITELIST_MODULE.get())
 			}
 			else if(item == SCContent.UNIVERSAL_BLOCK_MODIFIER.get())
 			{
-				world.destroyBlock(getPosition(), false);
-				Block.spawnAsEntity(world, getPosition(), getDisguiseModule());
-				Block.spawnAsEntity(world, getPosition(), getWhitelistModule());
+				if (!getDisguiseModule().isEmpty())
+				{
+					List<Block> blocks = ((ModuleItem)getDisguiseModule().getItem()).getBlockAddons(getDisguiseModule().getTag());
+
+					if(blocks.size() > 0)
+					{
+						Block block = blocks.get(0);

```

Comment:
Don't really need this variable here. You can just call `blocks.get(0)` directly in the if condition below.